src/sync.h

```diff
@@ -119,30 +127,37 @@ namespace librealsense
         const std::vector<stream_id>& get_streams() const override;
         const std::vector<rs2_stream>& get_streams_types() const override;
 
+        std::string get_name() const override;
     private:
-        std::vector<stream_id> _stream_id;
+
+        std::string _name;
         std::vector<rs2_stream> _stream_type;
     };
 
     class composite_matcher : public matcher
     {
     public:
-        composite_matcher(std::vector<std::shared_ptr<matcher>> matchers);
+        composite_matcher(std::vector<std::shared_ptr<matcher>> matchers, std::string name);
 
 
         virtual bool are_equivalent(frame_holder& a, frame_holder& b) { return true; };
         virtual bool is_smaller_than(frame_holder& a, frame_holder& b) { return true; };
         virtual bool skip_missing_stream(std::vector<matcher*> synced, matcher* missing) { return false; };
-        virtual void clean_dead_streams(frame_holder& f) = 0;
+        virtual void clean_inactive_streams(frame_holder& f) = 0;
+        virtual void update_last_arrived(frame_holder& f, matcher* m) = 0;
+
+        void dispatch(frame_holder f, syncronization_environment env) override;
         void sync(frame_holder f, syncronization_environment env) override;
         const std::vector<stream_id>& get_streams() const override;
         const std::vector<rs2_stream>& get_streams_types() const override;
         std::shared_ptr<matcher> find_matcher(const frame_holder& f);
 
+        std::string get_name() const override;
+
     private:
         std::vector<stream_id> _streams;
         std::vector<rs2_stream> _streams_type;
-
+        std::string _name;
```

Comment:
std::vector<rs2_stream> _streams_type;
 std::string _name;
Are already declared in the base matcher class